x-pack/plugin/runtime-fields/src/main/java/org/elasticsearch/xpack/runtimefields/mapper/ScriptFieldMapper.java

```diff
@@ -56,81 +73,118 @@ protected String contentType() {
+            "runtime_type",
+            true,
+            mapper -> toType(mapper).runtimeType,
+            null
+        );
+        // TODO script and runtime_type can be updated: what happens to the currently running queries when they get updated?
+        // do all the shards get a consistent view?
+        private final Parameter<Script> script = new Parameter<>("script", true, null,
+            Builder::parseScript, mapper -> toType(mapper).script);
         private final ScriptService scriptService;
 
-        private String runtimeType;
-        private Script script;
-
         protected Builder(String name, ScriptService scriptService) {
-            super(name, FIELD_TYPE);
+            super(name);
             this.scriptService = scriptService;
         }
 
-        public void runtimeType(String runtimeType) {
-            this.runtimeType = runtimeType;
-        }
-
-        public void script(Script script) {
-            this.script = script;
+        @Override
+        protected List<Parameter<?>> getParameters() {
+            return List.of(meta, runtimeType, script);
         }
 
         @Override
         public ScriptFieldMapper build(BuilderContext context) {
-            if (runtimeType == null) {
-                throw new IllegalArgumentException("runtime_type must be specified");
+            if (runtimeType.getValue() == null) {
+                throw new IllegalArgumentException("runtime_type must be specified for script field [" + name + "]");
             }
-            if (script == null) {
-                throw new IllegalArgumentException("script must be specified");
+            if (script.getValue() == null) {
+                throw new IllegalArgumentException("script must be specified for script field [" + name + "]");
             }
 
             MappedFieldType mappedFieldType;
-            if (runtimeType.equals("keyword")) {
-                StringScriptFieldScript.Factory factory = scriptService.compile(script, StringScriptFieldScript.CONTEXT);
-                mappedFieldType = new RuntimeKeywordMappedFieldType(buildFullName(context), script, factory, meta);
+            if (runtimeType.getValue().equals("keyword")) {
+                StringScriptFieldScript.Factory factory = scriptService.compile(script.getValue(), StringScriptFieldScript.CONTEXT);
+                mappedFieldType = new RuntimeKeywordMappedFieldType(buildFullName(context), script.getValue(), factory, meta.getValue());
             } else {
                 throw new IllegalArgumentException("runtime_type [" + runtimeType + "] not supported");
             }
-            // TODO copy to and multi_fields... not sure what needs to be done.
-            return new ScriptFieldMapper(name, mappedFieldType, multiFieldsBuilder.build(this, context), copyTo);
+            // TODO copy to and multi_fields should not be supported, parametrized field mapper needs to be adapted
```

Comment:
@romseygeek and this one too. I think I would prefer making the changes upstream instead of adapting parameterized field mapper in the feature branch. Thoughts?

Reply:
++, I can work on this one.

Reply:
thanks that would be great!